### PR TITLE
Move away from BuildConfig-backed jobs and various other cleanups

### PR DESCRIPTION
**Pull request by jlebon**
This *finally* moves us away from BuildConfig-backed Jenkins pipeline jobs which have been giving us a lot of issues. There's a lot here, but it's hopefully split into consumable commits. Along the way I did some other cleanups as I came by them. Can split things out into multiple PRs if desired.

See individual commits.

### CODE REVIEW

**Review by dustymabe — APPROVED** [review]
LGTM. This is such a big change should we tear down our existing project and start from scratch to prove the bringup steps still work?

File: HACKING.md [hunk 1]
```diff
@@ -286,10 +286,6 @@ circumstances. Here are some of them:
     - Git source URL and optional git ref for pipeline Jenkinsfile.
 - `--config <URL>[@REF]`
     - Git source URL and optional git ref for FCOS config.
-- `--pvc-size <SIZE>`
-    - Size of the cache PVC to create. Note that the PVC size cannot be
-      changed after creation. The format is the one understood by
-      Kubernetes, e.g. `30Gi`.
```
[the remark this answers is not in the document]
Comment (jlebon) [reply HACKING.md:292]: Yeah, doesn't feel like it's worth the complexity. Git will have it if we ever want to bring it back. :)

**Comment by jlebon** [comment]
> LGTM. This is such a big change should we tear down our existing project and start from scratch to prove the bringup steps still work?

Hmm yeah, that's probably a good call. Though maybe a halfway approach where we at least don't delete the secrets?

**Comment by dustymabe** [comment]
> Hmm yeah, that's probably a good call. Though maybe a halfway approach where we at least don't delete the secrets?

Works for me, though this might be a good chance to test the "delete project" ansible script and see if we actually can or not.

**Review by aaradhak — APPROVED** [review]
LGTM

**Pull request merged**